Objects/listobject.c

```diff
@@ -180,6 +180,22 @@ PyList_New(Py_ssize_t size)
     return (PyObject *) op;
 }
 
+static inline PyObject *
+list_new(Py_ssize_t size) {
```

Comment:
I'd give this function a name that makes it clear that the memory will not be initialised. Traditionally, `[builtin]_new` is the name of the `tp_new()` slot function, which this isn't.

Reply:
something like: list_new_uninitialized()? Or maybe list_new_prealloc()?

Reply:
I prefer `list_new_uninitialized` over `list_new_prealloc`. prealloc makes me think the function is to allocate more memory than needed now for later usage but it is not.